programs/ido-program/tests/ido-program.ts

```diff
@@ -0,0 +1,466 @@
+                toPubkey: project.publicKey,
+                lamports: 2000000000,
+            }),
+            SystemProgram.transfer({
+                fromPubkey: payer.publicKey,
+                toPubkey: investor.publicKey,
+                lamports: 2000000000,
+            })
+        );
+
+        await provider.sendAndConfirm(tx, [payer]);
+
+        fiatMint = await createMint(
+            provider.connection,
+            payer,
+            mintAuthority.publicKey,
+            undefined,
+            0,
+            undefined,
+            undefined,
+            TOKEN_PROGRAM_ID
+        );
+
+        nativeMint = await createMint(
+            provider.connection,
+            payer,
+            mintAuthority.publicKey,
+            undefined,
+            0,
+            undefined,
+            undefined,
+            TOKEN_PROGRAM_ID
+        );
+
+        projectFiat = await createAccount(
+            provider.connection,
+            payer,
+            fiatMint,
+            project.publicKey,
+            undefined,
+            undefined,
+            TOKEN_PROGRAM_ID
+        );
+
+        projectNative = await createAccount(
+            provider.connection,
+            payer,
+            nativeMint,
+            project.publicKey,
+            undefined,
+            undefined,
+            TOKEN_PROGRAM_ID
+        );
+
+        investorFiat = await createAccount(
+            provider.connection,
+            payer,
+            fiatMint,
+            investor.publicKey,
+            undefined,
+            undefined,
+            TOKEN_PROGRAM_ID
+        );
+
+        investorNative = await createAccount(
+            provider.connection,
+            payer,
+            nativeMint,
+            investor.publicKey,
+            undefined,
+            undefined,
+            TOKEN_PROGRAM_ID
+        );
+
+        await mintTo(
+            provider.connection,
+            payer,
+            nativeMint,
+            projectNative,
+            mintAuthority,
+            nativeTokenAmount.toNumber()
+        );
+
+        await mintTo(
+            provider.connection,
+            payer,
+            fiatMint,
+            investorFiat,
+            mintAuthority,
+            10000
+        );
+
+        const projectNativeTokenAccount = await getAccount(
+            provider.connection,
+            projectNative
+        );
+
+        assert.strictEqual(
+            projectNativeTokenAccount.amount.toString(),
+            nativeTokenAmount.toNumber().toString()
+        );
+    });
+
+    it("Can initialize the Pool", async () => {
+        const [_poolSigner, bump] =
+            anchor.web3.PublicKey.findProgramAddressSync(
+                [nativeMint.toBuffer()],
+                program.programId
+            );
+
+        poolSigner = _poolSigner;
+
+        redeemableMint = await createMint(
+            provider.connection,
+            payer,
+            poolSigner,
+            undefined,
```

Comment:
This parameter is type `PublicKey | null`
```suggestion
            null,
```